agent/asm/asm_test.go

```diff
@@ -189,3 +190,51 @@ func createASMInterface(secretValue string) mockGetSecretValue {
 		},
 	}
 }
+
+type brokenGetSecretValue struct {
+	secretsmanageriface.SecretsManagerAPI
+}
+
+func (m brokenGetSecretValue) GetSecretValue(input *secretsmanager.GetSecretValueInput) (*secretsmanager.GetSecretValueOutput, error) {
+	return nil, errors.New("ResourceNotFoundException: Secrets Manager can't find the specified secret.")
+}
+
+func TestGetDockerAuthFromASMErrorMessage(t *testing.T) {
+	origError := errors.New("ResourceInitializationError: The task can't retrieve the secret with ARN 'secret-value-id' from AWS Secrets Manager. Check that the secret ARN is correct. ResourceNotFoundException: Secrets Manager can't find the specified secret.")
+	expectedErr := ASMError{FromError: origError}
+
+	asmClient := brokenGetSecretValue{}
```

Comment:
nit: we use [gomock](https://github.com/golang/mock) for mocking in Agent codebase. You may want to use it instead of defining a new implementation of `secretsmanageriface.SecretsManagerAPI` for consistency's sake.

Reply:
I updated this test to use gomock